riscv-page-tables/src/page_table.rs

```diff
@@ -714,6 +754,34 @@ impl<'a, T: FirstStagePagingMode> FirstStageMapper<'a, T> {
         let mut inner = self.owner.inner.lock();
         inner.map_4k_leaf(vaddr, paddr, pte_perms)
     }
+
+    /// Maps `vaddr` to `paddr`, The caller must guarantee that paddr points to a page it is safe to
+    /// map in this page table.
+    ///
+    /// This functions works like map_4k_addr, but takes a page size parameter.
+    /// # Safety
+    ///
+    /// Don't create aliases.
+    pub unsafe fn map_general_addr(
```

Comment:
Should `map_4k_addr` be changed to call `map_general_addr` with PageSize4k specified? Also, `map_addr` instead of `map_general_addr`.

Reply:
fixed